packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -100,119 +124,132 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
     initialize(): void {
         this.initializeProviders();
     }
-    protected async initializeProviders(): Promise<void> {
+
+    protected initializeProviders(): void {
         try {
-            const providers = this.createProviders();
-            this.toDispose.pushAll(providers);
-            await Promise.all(providers.map(p => p.ready));
+            this.createProviders();
             if (this.toDispose.disposed) {
                 return;
             }
-            this.providers.push(...providers);
-            for (const provider of providers) {
-                provider.onDidPreferencesChanged(_ => this.reconcilePreferences());
+            for (const [scope, provider] of this.providersMap.entries()) {
+                if (scope !== PreferenceScope.DEFAULT) {
+                    this.toDispose.push(provider.onDidPreferencesChanged(changes =>
+                        this.reconcilePreferences(changes)
+                    ));
+                }
             }
-            this.reconcilePreferences();
-            this._ready.resolve();
+            Promise.all(this.providers.map(p => p.ready)).then(() => this._ready.resolve());
```

Comment:
what about `this._ready` when one of `p.ready` throws?

Reply:
in the current code (both on master and this PR), p.ready from the preference-providers [never rejects](https://github.com/theia-ide/theia/blob/master/packages/preferences/src/browser/abstract-resource-preference-provider.ts#L50), and therefore we don't take care of the scenario you are referring to.